extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelProcessor.java

```diff
@@ -346,4 +347,10 @@ void unremovableBeans(BuildProducer<AdditionalBeanBuildItem> beanProducer) {
                 .map(CamelRoutesBuilderClassBuildItem::new)
                 .collect(Collectors.toList());
     }
+
+    @BuildStep
+    NativeImageResourceBuildItem initResources() {
+        return new NativeImageResourceBuildItem(
+                "META-INF/services/org/apache/camel/bean-processor-factory");
```

Comment:
@davsclaus I've returned `getBootstrapFactoryFinder`. To make it work (I've tested it successfully with a few integration tests locally) , I had to add this resource int camelProcessor. This solution does not seems correct to me. @lburgazzoli  May I ask you for help? (I see see you in annotation in this class)

Reply:
Oh that seems correct as there are a bunch of files in those folders for various Camel components that are discovered via classpath/factory-finder and so quarkus needs to ensure those meta files are included.

Reply:
great, thanks for confirmation, let see, if all tests are green.

Reply:
afaik those files are scanned and the related classes are added to the FastFactoryFinder so it should not be required to include the resource, see: https://github.com/apache/camel-quarkus/blob/master/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelProcessor.java#L136-L159

Reply:
@lburgazzoli  yes, I've noticed that, therefore I'm still thinking that my change is unnecessary, but it fixes problem. But I'm not sure why it is required...

Reply:
btw, I think it is fine to have this and open an issue for further improvements

Reply:
I've created issue https://github.com/apache/camel-quarkus/issues/1995